enzyme/test/Enzyme/ProbProg/simple-condition.ll

```diff
@@ -0,0 +1,92 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -S | FileCheck %s
+
+@enzyme_condition = global i32 0
+
+@.str = private constant [3 x i8] c"mu\00"
+@.str.1 = private constant [2 x i8] c"x\00"
+
+declare double @normal(double, double)
+declare double @normal_logpdf(double, double, double)
+
+declare i8* @__enzyme_newtrace()
+declare void @__enzyme_freetrace(i8*)
+declare i8* @__enzyme_get_trace(i8*, i8*)
+declare i64 @__enzyme_get_choice(i8*, i8*, i8*, i64)
+declare void @__enzyme_insert_call(i8*, i8*, i8*)
+declare void @__enzyme_insert_choice(i8* %trace, i8*, double, i8*, i64)
+declare i1 @__enzyme_has_call(i8*, i8*)
+declare i1 @__enzyme_has_choice(i8*, i8*)
+declare double @__enzyme_sample(double (double, double)*, double (double, double, double)*, i8*, double, double)
+declare i8* @__enzyme_trace(void ()*)
+declare i8* @__enzyme_condition(void ()*, i32, i8*)
+
+define void @test() {
+entry:
+  %mu = call double @__enzyme_sample(double (double, double)* @normal, double (double, double, double)* @normal_logpdf, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @.str, i64 0, i64 0), double 0.0, double 1.0)
+  %x = call double @__enzyme_sample(double (double, double)* @normal, double (double, double, double)* @normal_logpdf, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @.str.1, i64 0, i64 0), double %mu, double 1.0)
+  ret void
+}
+
+define i8* @generate() {
+entry:
+  %call = call i8* @__enzyme_trace(void ()* @test)
+  ret i8* %call
+}
+
+define i8* @condition(i8* %trace) {
+entry:
+  %0 = load i32, i32* @enzyme_condition
+  %call = call i8* @__enzyme_condition(void ()* @test, i32 %0, i8* %trace)
+  ret i8* %call
+}
+
+
+; CHECK: define internal i8* @condition_test(i8* %trace)
+; CHECK-NEXT: entry:
+; CHECK-NEXT:   %trace1 = call i8* @__enzyme_newtrace()
+; CHECK-NEXT:   %0 = call i1 @__enzyme_has_choice(i8* %trace, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @.str, i64 0, i64 0))
+; CHECK-NEXT:   br i1 %0, label %condition.mu.with.trace, label %condition.mu.without.trace
+
```

Comment:
Can we have this not run the inlining during Enzyme, and instead look at the gnereated test function?

Reply:
Which function should not be inlined? I don't see any inlining happening here.

Reply:
The augmented test function. Like This should be a call to some sort of conditioned_test(...)

Reply:
Like this?
```
define i8* @condition(i8* %trace) {
  %0 = load i32, i32* @enzyme_condition, align 4
  %1 = call i8* @condition_test(i8* %trace)
  ret i8* %1
}
```

Reply:
Yeah

Reply:
done